wee_slack.py

```diff
@@ -2605,6 +2605,23 @@ def process_message(message_json, eventrouter, store=True, **kwargs):
             channel.store_message(message, team)
         dbg("NORMAL REPLY {}".format(message_json))
 
+    download_location = config.get_string('files_download_location')
+    if not download_location:
+        return
+
+    for f in message_json.get('files', []):
+        if f.get('mode', '') == 'tombstone':
+            continue
+
+        file_name = '{}_{}.{}'.format(f['id'], f['title'], f['filetype'])
```

Comment:
The id is just a random string, not very useful to the user. I guess you include it in case there are two files with the same title, but I think at least we should place the title first.

The title may include the file type, so we should only append the file type if it does not.

Reply:
If the title has a file extension or not depends on the upload. If you upload an existing image and don't change the title, it will have a file extension in my experience.

If the id goes after the title as I suggested, I guess it doesn't matter that much if we remove the file extension or not.

I do wonder if we should drop the id though. It would be much nicer to just use the title which will be equal to the original file name if it isn't changed. Other programs usually append a number at the end if the file name already exists, so we could do the same.

Reply:
I see several with the title `Image from iOS`. If we rewrite the URL displayed in weeslack to file:// (or perhaps better configurable), then I think dropping the ID would be fine. Otherwise, you'd be hard pressed to find what you want among files with such titles :)

Reply:
Hm, right, that not that useful. I think it would be easier to find it with a number than with the id though. With a number you can just take the latest. Looking through lots of id's would be hard.

Rewriting the link if the option is enabled seems like a good idea.